packages/components/src/modal/test/index.js

```diff
@@ -46,4 +46,15 @@ describe( 'Modal', () => {
 		const titleId = within( dialog ).getByText( 'Test Title' ).id;
 		expect( dialog ).toHaveAttribute( 'aria-labelledby', titleId );
 	} );
+
+	it( 'hides the header when the `__experimentalHideHeader` prop is used', () => {
+		render(
+			<Modal title="Test Title" __experimentalHideHeader={ true }>
```

Comment:
Seems like we could ditch the `={ true }`:

```suggestion
			<Modal title="Test Title" __experimentalHideHeader>
```